interact_Signin.js

```diff
@@ -0,0 +1,132 @@
+import { chkLocal } from "/interact.js";
+
+chkLocal();
+
+//이메일 input에서 focus out 할 때, 값이 없을 경우 아래에 “이메일을 입력해 주세요.”
+//이메일 input에서 focus out 할 때, 이메일 형식에 맞지 않는 값이 있는 경우 아래에 “올바른 이메일 주소가 아닙니다.”
+
+const inputel=document.querySelector('#emailinput'); //이메일 input 박스
+const emailframe=document.querySelector('#emailframe'); //이메일 input 박스 frame
+const errmsg=document.createElement('p'); //에러메시지
+const emailtest = /^[^\s@]+@[^\s@]+\.[^\s@]+$/; //이메일 테스팅 코드
+
+function emaininput(e){
+  if(e.target.value==''){
+    errmsg.textContent='이메일을 입력해 주세요.';
+    emailframe.append(errmsg);
+    console.log(1);
+    console.log(e.target.type);
+  }
+  else{
+    if(!emailtest.test(e.target.value)){ //이메일 확인 테스트
+      errmsg.textContent='올바른 이메일 주소가 아닙니다.';
+      emailframe.append(errmsg);
+    }
+    else{
+      if(errmsg.textContent){ //만약에 올바른 이메일 주소를 작성하였으면 에러메시지 삭제
+        errmsg.remove();
+      }
+      //console.log(2);
+    }
+    
+  }
```

Comment:
r: early return 사용해 리팩터링하시면 위의 조건에서 errmsg.textContent를 할당했는지를 검사하지 않아도 될 것 같아요.

errmsg.textContent에 에러메시지가 들어갔는지 확인해 이메일 주소의 유효성을 검증하는 방법은 직관적이지 않고, 조건이 늘어났을 때 예측이 어려워보입니다.

```suggestion
  } else if (!emailtest.test(e.target.value)) { //이메일 확인 테스트
    errmsg.textContent='올바른 이메일 주소가 아닙니다.';
    emailframe.append(errmsg);
    
    /**
     * early return으로 함수 종료
     */
    return;
  }

  /**
   * early return으로 종료되지 않고 끝까지 접근했으므로 유효한 이메일, 에러메시지 삭제
   */
  errmsg.remove();
  }
```